services/harmony/app/backends/workflow-orchestration/work-item-updates.ts

```diff
@@ -966,19 +1032,23 @@ export async function handleBatchWorkItemUpdatesWithJobId(
   // group updates by workflow step index to make sure at least one completion check is performed for each step
   const groups = _.groupBy(updates, (update) => update.update.workflowStepIndex);
   for (const workflowStepIndex of Object.keys(groups)) {
-    const nextWorkflowStep = await (await logAsyncExecutionTime(
-      getWorkflowStepByJobIdStepIndex,
-      'HWIUWJI.getWorkflowStepByJobIdStepIndex',
-      logger))(db, jobID, parseInt(workflowStepIndex) + 1);
-
-    const preprocessedWorkItems: WorkItemUpdateQueueItem[] = await Promise.all(
-      groups[workflowStepIndex].map(async (item: WorkItemUpdateQueueItem) => {
-        const { update, operation } = item;
-        const result = await preprocessWorkItem(update, operation, logger, nextWorkflowStep);
-        item.preprocessResult = result;
-        return item;
-      }));
-    await processWorkItems(jobID, parseInt(workflowStepIndex), preprocessedWorkItems, logger);
+    if (groups[workflowStepIndex][0].update.message_category === 'granValidation') {
```

Comment:
I think `granValidation` should be a `const` string defined somewhere or better yet an enumerated type.

Reply:
I will punt this for now -- hard to find a good location for this constant.